arm64: dts: qcom: msm8939-asus-z00t: add battery and hall sensor

erikas9987 opened the pull request:
Asus ZenFone 2 Laser/Selfie includes a battery measured by voltage mode BMS and a simple GPIO hall effect sensor. This commit includes support for these components.

TravMurav commented:
These changes look good to me, since the base device is upstream and those changes are upstreamable too, you may want to split them up to "logical changes" (i.e. for upstream I think the usual workflow for the first patch is to cram as much as possible into "add a new device" patch, and subsequent changes would be separate for each change), i.e. here one for hall sensor and one for bms, or 3 changes with a small fix for the buttons to add the label separate, at your discretion. (if you combine them, may want to motivate that in the commit message, "while at it, fix the missing label blah blah")

(...) With that I will be happy to merge those patches as FROMLIST: when you send v1

erikas9987 commented:
> you may want to split them up to "logical changes" 

Like this?

TravMurav commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -49,6 +49,8 @@
 		pinctrl-0 = <&gpio_hall_sensor_default>;
 		pinctrl-names = "default";
 
+		label = "GPIO Hall Effect Sensor";
```
I think it makes more sense to have this inside the commit that adds the node, and have the other one as a "fix" for the old node. if you want, you can even add a Fixes: tag to this commit (targeting the "add device" one), though this change is probably trivial enough to not warrant it.

a-andre commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -20,16 +20,55 @@
 	};
 
+	gpio-hall-sensor {
+		compatible = "gpio-keys";
```
Not sure if it is still true. But I was asked to combine the `gpio-keys` nodes.

https://lore.kernel.org/all/a1563e29-2f20-4b1e-acbf-2b6d77dc7831@linaro.org/

erikas9987 replied:
Is the current order correct, or should I order by `gpios` property value instead?

TravMurav replied:
I think in this case, alphabetical is correct, sorting by pins/gpios is usually done for huge tables like pinctrl where "pins" themselves are the primary information (though even in that case we usually sort pinctrl by names since we don't define the whole table but a tiny subset in our dts)

> 2. Nodes without unit addresses shall be ordered alpha-numerically by the node
>   name.  For a few node types, they can be ordered by the main property, e.g.
>   pin configuration states ordered by value of "pins" property.

https://elixir.bootlin.com/linux/v6.14-rc4/source/Documentation/devicetree/bindings/dts-coding-style.rst#L56-L58


(TBH i'd argue that a set of keys is a separate device from a hall sensor but that's petty semantics)

erikas9987 committed:
FROMLIST: arm64: dts: qcom: msm8939-asus-z00t: add battery

This device tracks remaining battery capacity percentage using voltage
mode BMS. This commit enables the pm8916_bms node and adds a battery
node to track its capacity. Battery properties were taken from the
downstream code and from the information printed on the battery itself.

Link: https://lore.kernel.org/r/20251115-battery-hall-v1-1-1586283d17c7@gmail.com
Signed-off-by: Erikas Bitovtas <xerikasxx@gmail.com>

erikas9987 committed:
FROMLIST: arm64: dts: qcom: msm8939-asus-z00t: add hall sensor

This device uses ANPEC APX9131 hall sensor. It is a basic GPIO hall
sensor for which a generic "gpio-keys" device tree node configuration
suffices.

Link: https://lore.kernel.org/r/20251115-battery-hall-v1-2-1586283d17c7@gmail.com
Signed-off-by: Erikas Bitovtas <xerikasxx@gmail.com>

TravMurav commented:
Thanks!

FTR https://lore.kernel.org/linux-arm-msm/20251115-battery-hall-v1-0-1586283d17c7@gmail.com/
